src/components/form/fields/FileUploadField.tsx

```diff
@@ -48,10 +85,9 @@ export function FileUploadField({
             <CustomFileUploadField
               field={{
                 ...field,
-                onChange: (value: any) => {
-                  field.onChange(value);
-                  onChange?.(value);
-                },
+                value: null,
+                onChange: (value: any) =>
+                  handleFileChange(value, field, multiple),
```

Comment:
I think we can simplify `handleFileChange` function signature and only pass it the value (with this change the function should be renamed to `convertFilesToBase64`)

```suggestion
                onChange: async (value: any) => {
                  const fileUploadValue = await convertFilesToBase64(value);
                  field.onChange(fileUploadValue);
                  onChange?.(fileUploadValue); 
                }
```